lib/datasets/ames-housing.rb

```diff
@@ -0,0 +1,131 @@
+                        :land_slope,
+                        :neighborhood,
+                        :condition_1,
+                        :condition_2,
+                        :bldg_type,
+                        :house_style,
+                        :overall_qual,
+                        :overall_cond,
+                        :year_built,
+                        :year_remod_add,
+                        :roof_style,
+                        :roof_matl,
+                        :exterior_1st,
+                        :exterior_2nd,
+                        :mas_vnr_type,
+                        :mas_vnr_area,
+                        :exter_qual,
+                        :exter_cond,
+                        :foundation,
+                        :bsmt_qual, 
+                        :bsmt_cond,
+                        :bsmt_exposure,
+                        :bsmt_fin_type_1,
+                        :bsmt_fin_sf_1,
+                        :bsmt_fin_type_2,
+                        :bsmt_fin_sf_2,
+                        :bsmt_unf_sf,
+                        :total_bsmt_sf,
+                        :heating,
+                        :heating_qc,
+                        :central_air,
+                        :electrical,
+                        :first_flr_sf,
+                        :second_flr_sf,
+                        :low_qual,
+                        :fin_sf,
+                        :gr_liv_area,
+                        :bsmt_full_bath,
+                        :bsmt_half_bath,      
+                        :full_bath,
+                        :half_bath,
+                        :bedroom_abv_gr,
+                        :kitchen_abv_gr,
+                        :kitchen_qual,
+                        :tot_rms_abv_grd,
+                        :functional,
+                        :fireplaces,
+                        :fireplace_qu,
+                        :garage_type,
+                        :garage_yr_blt,
+                        :garage_finish,
+                        :garage_cars,
+                        :garage_area,
+                        :garage_qual,
+                        :garage_cond,
+                        :paved_drive,
+                        :wood_deck_sf,
+                        :open_porch_sf,
+                        :enclosed_porch,
+                        :three_ssn_porch,
+                        :screen_porch,
+                        :pool_area,
+                        :pool_qc,
+                        :fence,
+                        :misc_feature,
+                        :misc_val,
+                        :mo_sold, 
+                        :yr_sold,
+                        :sale_type,
+                        :sale_condition,
+                        :sale_price)
+      
+
+    def initialize
+      super()
+      @metadata.id = "ames-housing"
+      @metadata.name = "Ames Housing"
+      @metadata.url = "http://jse.amstat.org/v19n3/decock/DataDocumentation.txt"
```

Comment:
Maybe should we use [house_prices datasets](https://www.openml.org/search?type=data&sort=runs&id=42165&status=active) in OpenML because [sklearn.datasets.load_boston](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html) mentioned it [here](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html#id4)?
If we use this data, there is one problem. This dataset file is `.arff`. So we have to implement parser function about it.

Reply:
Thanks for your comments. The data is at https://api.openml.org/data/v1/download/21754539/house_prices.arff , the json and xml files contain metadata only.

arff is used in Weka, there are readers for [R](https://rdrr.io/cran/farff/src/R/readARFF.R) and [Python](https://pythonhosted.org/liac-arff/), so such a reader may be helpful whether it is used here or not.

The data use in sklearn has been cleaned, but is in csv form.

As explained in https://www.tmwr.org/ames.html the data in the model data package has been transformed more than the data used in sklearn, with latitude and longitude information that is helpful for visualization.  It seems higher quality, but is available in rda format https://github.com/topepo/AmesHousing/tree/master/data though it is possible to export to csv format as explained at https://stackoverflow.com/questions/4487065/convert-rda-to-csv  An rda reader would likely also be helpful.

At the moment, all data is downloaded from external repositories which may disappear.  It may be good to host some of the data at a location that has better control, but probably this can be a separate issue.

Similarly plugin readers for rda and arff similar to https://github.com/red-data-tools/red-datasets-pandas should be separate issues.

Reply:
> Thanks for your comments. The data is at https://api.openml.org/data/v1/download/21754539/house_prices.arff , the json and xml files contain metadata only.

Sorry I thought there were Ames housing too. But it wasn't. Thank you!

---

Thank you for explaining the situation in details.
Now I agree with you to use [this data resource](http://jse.amstat.org/v19n3/decock/AmesHousing.txt). And it looks better way.

Reply:
Thanks for the explanation.
I agree with using http://jse.amstat.org/v19n3/decock/AmesHousing.txt (the original data) is suitable. I confirmed https://www.tandfonline.com/doi/citedby/10.1080/10691898.2011.11889627 too because http://jse.amstat.org/ was archived and move to https://tandfonline.com/toc/ujse20/current . It seems that new site https://www.tandfonline.com/doi/citedby/10.1080/10691898.2011.11889627 doesn't provide the original data. Other article such as https://www.tandfonline.com/doi/full/10.1080/26939169.2022.2074923 has the "Supplemental" tab but https://www.tandfonline.com/doi/citedby/10.1080/10691898.2011.11889627 doesn't have the tab.